### PR TITLE
upgrade to Baragon 0.1.4 + handle INVALID_REQUEST_NOOP

### PR DESCRIPTION
**Pull request by tpetr**
@wsorenson @ssalinas could you help verify this logic is accurate?

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java [hunk 1]
```diff
@@ -300,6 +300,7 @@ private CheckTaskState getTaskState(SingularityTask task) {
   private Optional<CheckTaskState> checkLbState(BaragonRequestState lbState) {
     switch (lbState) {
       case SUCCESS:
+      case INVALID_REQUEST_NOOP:
         return Optional.of(CheckTaskState.HEALTHY);
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java]: Is a task healthy if baragon says the lb state is INVALID?

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java [hunk 2]
```diff
@@ -383,6 +383,7 @@ private boolean shouldRemoveLbState(SingularityTaskId taskId, SingularityLoadBal
       case UNKNOWN:
       case WAITING:
       case SUCCESS:
+      case INVALID_REQUEST_NOOP:
```
Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityCleaner.java]: In this case, we don't need to remove from Baragon b/c Baragon doesn't know about it, right?

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java (hunk 1 above)
Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java]: in my original notes, INVAILD_REQUEST_NOOP should be fired when removing a nonexistent upstream, or removing a service from a nonexistent LB -- let's sync up if that state is supposed to mean something else

Comment (wsorenson) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java]: And if we get it back from an ADD request, I can't see the client handling that as nominal

Comment (tpetr) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityNewTaskChecker.java]: right -- good point

**Comment by ssalinas** [comment]
@tpetr added a condition for `INVALID_REQUEST_NOOP` to go to a `FAILED` deploy state in deploy checker, imitating the behavior of a `FAILED` load balancer request

**Pull request merged**